chat-plugins/modified.js

```diff
@@ -32,93 +32,48 @@ function parseStatus(text, encoding) {
+			if (ranks.includes(rank)) {
 				let name = Users.usergroups[u].substr(1);
-				persons.push({
-					name: name,
-					rank: rank,
-				});
+				if (!rankLists[rank]) rankLists[rank] = [];
+				if (name) rankLists[rank].push(SG.nameColor(name, (Users(name) && Users(name).connected)));
 			}
 		}
-		let staff = {
-			"admins": [],
-			"leaders": [],
-			"bots": [],
-			"mods": [],
-			"drivers": [],
-			"voices": [],
-		};
-		persons = persons.sort((a, b) => toId(a.name).localeCompare(toId(b.name))); // No need to return, arrow functions with single lines have an implicit return
-		function nameColor(name) {
-			if (Users.getExact(name) && Users(name).connected) {
-				return '<b><i><font color="' + hashColorWithCustoms(name) + '">' + Chat.escapeHTML(Users.getExact(name).name) + '</font></i></b>';
-			} else {
-				return '<font color="' + hashColorWithCustoms(name) + '">' + Chat.escapeHTML(name) + '</font>';
-			}
-		}
-		for (let j = 0; j < persons.length; j++) {
-			let rank = persons[j].rank;
-			let person = persons[j].name;
-			switch (rank) {
-			case '~':
-				staff['admins'].push(nameColor(person));
-				break;
-			case '&':
-				staff['leaders'].push(nameColor(person));
-				break;
-			case '*':
-				staff['bots'].push(nameColor(person));
-				break;
-			case '@':
-				staff['mods'].push(nameColor(person));
-				break;
-			case '%':
-				staff['drivers'].push(nameColor(person));
-				break;
-			case '+':
-				staff['voices'].push(nameColor(person));
-				break;
-			default:
-				continue;
 
-			}
-		}
-		connection.popup('|html|' +
-			'<h3>SpacialGaze Authority List</h3>' +
-			'<b><u>~Administrators (' + staff['admins'].length + ')</u></b>:<br />' + staff['admins'].join(', ') +
-			'<br />' +
-			'<br /><b><u>&Leaders (' + staff['leaders'].length + ')</u></b>:<br />' + staff['leaders'].join(', ') +
-			'<br />' +
-			'<br /><b><u>*Bots (' + staff['bots'].length + ')</u></b><br />' + staff['bots'].join(', ') +
-			'<br />' +
-			'<br /><b><u>@Moderators (' + staff['mods'].length + ')</u></b>:<br />' + staff['mods'].join(', ') +
-			'<br />' +
-			'<br /><b><u>%Drivers (' + staff['drivers'].length + ')</u></b>:<br />' + staff['drivers'].join(', ') +
-			'<br />' +
-			'<br /><b><u>+Voices (' + staff['voices'].length + ')</u></b>:<br />' + staff['voices'].join(', ') +
-			'<br /><br /><blink>(<b>Bold</b> / <i>Italic</i> = Currently Online)</blink>'
+		let buffer = Object.keys(rankLists).sort((a, b) =>
+			(Config.groups[b] || {rank: 0}).rank - (Config.groups[a] || {rank: 0}).rank
+		).map(r =>
+			(Config.groups[r] ? "<b>" + Config.groups[r].name + "s</b> (" + r + ")" : r) + ":\n" + rankLists[r].sort((a, b) => toId(a).localeCompare(toId(b))).join(", ")
 		);
+
+		if (!buffer.length) return connection.popup("This server has no global authority.");
+		connection.send("|popup||html|" + buffer.join("\n\n"));
 	},
 
 	'!roomauth': true,
 	roomstaff: 'roomauth',
-	roomauth1: 'roomauth',
```

Comment:
At that point, I did not know main was the one who added `/roomauth1`. I Apologize for any confusion.